vm/src/vm/vm_memory/memory_segments.rs

```diff
@@ -15,7 +15,7 @@ use crate::{
 pub struct MemorySegmentManager {
     pub segment_sizes: HashMap<usize, usize>,
     pub segment_used_sizes: Option<Vec<usize>>,
-    pub(crate) memory: Memory,
+    pub memory: Memory,
```

Comment:
this should not be made public.

Reply:
Some of the macros in `test_utils` require access to this field. As the goal is to make the macros usable out of tree, the easy solution is to make this field public. I understand the concern though. Any suggestion on how to work around the issue?

Reply:
> Any suggestion on how to work around the issue?

It depends on what you want to do in the tests. Maybe you could just use the `insert` family of functions.